### PR TITLE
Add a template release checklist, refs #3185

### PR DESCRIPTION
**Pull request by hdgarrood**
The intention is that this file can serve as a template / starting point
for each future release.

### CODE REVIEW

**Comment by JordanMartinez** [comment]
Maybe including a general visual dependency graph would help, too? It wouldn't necessarily list every step, but it would include each 'stage' of things that need to be done before anything else gets done.

**Comment by hdgarrood** [comment]
I don't think that's necessary; the people who are actually working through these steps would probably know what order they need to be done in.

**Comment by JordanMartinez** [comment]
For that audience (core contributors), that makes sense. However, there's a larger issue I have in mind, but I don't have all the details for it yet. Still, this is the wrong place to have that discussion. Sorry about that :sweat_smile:

File: RELEASE_CHECKLIST.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- [ ] ide plugins
+
+## JSON formats
+
+Have any of the following JSON formats changed? If so, it may be worth
+considering what effects this may have:
+
+- [ ] Corefn
+- [ ] JSON produced by `purs publish`
+  - [ ] check whether this affects Pursuit
```
Comment (kRITZCREEK) [inline RELEASE_CHECKLIST.md]: The IDE protocol is another JSON format that might break (though the affected users are a smaller group as it's primarily plugin authors)

**Pull request merged**